pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -688,20 +688,17 @@ func (r *ReconcileCredentialsRequest) Reconcile(ctx context.Context, request rec
 
 		var syncErr error
 		syncErr = r.CreateOrUpdateOnCredsExist(ctx, credsExists, syncErr, cr)
```

Comment:
When CR.spec.providerSpec.stsIAMRoleARN is unset, syncErr is still nil
https://github.com/openshift/cloud-credential-operator/blob/859de532440846fabddf5dd5a07417a45b47b7fb/pkg/aws/actuator/actuator.go#L342-L345 

So we fall into the else branch below where we call
https://github.com/openshift/cloud-credential-operator/blob/859de532440846fabddf5dd5a07417a45b47b7fb/pkg/operator/credentialsrequest/credentialsrequest_controller.go#L724

In other words CR.status.provisioned is still set to true when CR.spec.providerSpec.stsIAMRoleARN is unset and there is no Secret provisioned.

Reply:
The comment above aligns with what I observed today in a test (on this PR).

Reply:
Thanks! I think my next commit fixes it then.